crates/nu-protocol/src/config/mod.rs

```diff
@@ -219,374 +162,333 @@ impl Value {
-                                            report_invalid_value("should be an int", span, &mut errors);
+                                },
+                                "index_mode" => {
+                                    process_string_enum(
+                                        &mut config.table.index_mode,
+                                        &[key, key2],
+                                        value,
+                                        &mut errors
+                                    );
+                                }
+                                "trim" => {
+                                    match try_parse_trim_strategy(value, &mut errors) {
+                                        Ok(v) => config.table.trim = v,
+                                        Err(e) => {
+                                            // try_parse_trim_strategy() already adds its own errors
+                                            errors.push(e);
+                                            *value = config.table.trim.clone().into_value(span);
                                         }
                                     }
-                                    _ => {
-                                        report_invalid_key(&[key, key2], span, &mut errors);
-                                        return false
+                                }
+                                "show_empty" => {
+                                    process_bool_config(value, &mut errors, &mut config.table.show_empty);
+                                }
+                                "abbreviated_row_count" => {
+                                    if let Ok(b) = value.as_int() {
+                                        if b < 0 {
+                                            report_invalid_value("should be an int unsigned", span, &mut errors);
+                                        }
+
+                                        config.table.abbreviated_row_count = Some(b as usize);
+                                    } else {
+                                        report_invalid_value("should be an int", span, &mut errors);
                                     }
-                                };
-                                true
-                             });
-                        } else {
-                            report_invalid_value("should be a record", span, &mut errors);
-                            // Reconstruct
-                            *value = Value::record(
-                                record! {
-                                    "mode" => config.table_mode.reconstruct_value(span),
-                                    "index_mode" => config.table_index_mode.reconstruct_value(span),
-                                    "trim" => reconstruct_trim_strategy(&config, span),
-                                    "show_empty" => Value::bool(config.table_show_empty, span),
-                                },
```

Comment:
Definite improvement to move to `IntoValue` here. When refactoring in #10896 I worked on all the subpoints above but missed that some of them weren't part of the previous record creation. Now this is kept in check by the respective config struct.